src/OpenTK.Graphics/OpenGL/GL.Manual.cs

```diff
@@ -5,34 +5,58 @@ namespace OpenTK.Graphics.OpenGL
 {
     // FIXME: Remove this when it's fixed
     // This is here because there in the gl.xml
-    // one of the paramteres for "glSampleMaskIndexedNV"
+    // one of the parameters for "glSampleMaskIndexedNV"
     // is marked with a group named this, but this group is never referenced
     // anywhere else in the file.
     public enum SampleMaskNV
     { }
 
     public static unsafe partial class GL
     {
-        public static void ShaderSource(uint shader, string str)
+        public static void ShaderSource(uint shader, string shaderText)
         {
-            IntPtr str_iptr = Marshal.StringToCoTaskMemAnsi(str);
-            int length = str.Length;
-            GL.ShaderSource(shader, 1, (byte**)&str_iptr, length);
-            Marshal.FreeCoTaskMem(str_iptr);
+            if (string.IsNullOrEmpty(shaderText))
+            {
+                throw new ArgumentNullException(nameof(shaderText));
+            }
+
+            var shaderTextPtr = Marshal.StringToCoTaskMemAnsi(shaderText);
+            var length = shaderText.Length;
+            ShaderSource(shader, 1, (byte**)&shaderTextPtr, length);
+            Marshal.FreeCoTaskMem(shaderTextPtr);
         }
 
         public static void GetShaderInfoLog(uint shader, out string info)
         {
             int length = default;
-            GL.GetShader(shader, ShaderParameterName.InfoLogLength, ref length);
+            GetShader(shader, ShaderParameterName.InfoLogLength, ref length);
             if (length == 0)
             {
                 info = string.Empty;
             }
             else
             {
-                GL.GetShaderInfoLog(shader, length, ref length, out info);
+                GetShaderInfoLog(shader, length, ref length, out info);
             }
         }
+
+        /// <summary>
+        /// Create a stand-alone program from an array of null-terminated source code strings
+        /// </summary>
+        /// <param name="shaderType">Specifies the type of shader to create</param>
+        /// <param name="count">Specifies the number of source code strings in the array strings</param>
+        /// <param name="shaderText"></param>
+        /// <exception cref="ArgumentNullException"></exception>
+        public static void CreateShaderProgramv(ShaderType shaderType, int count, string shaderText)
```

Comment:
This should be `CreateShaderProgram` to match the naming convention of opentk.

Reply:
Ok, I can change it, was going with the existing name for it.

Reply:
Fixed.